rust/connlib/tunnel/src/peer.rs

```diff
@@ -180,6 +275,41 @@ impl ClientOnGateway {
         self.recalculate_filters();
     }
 
+    pub(crate) fn poll_event(&mut self) -> Option<GatewayEvent> {
+        self.buffered_events.pop_front()
+    }
+
+    pub(crate) fn expire_nat(&mut self, now: Instant) {
+        let expired_translations = self
+            .permanent_translations
+            .values()
+            .filter(|(_, _, _, _, slated_for_refresh)| *slated_for_refresh);
+        let mut for_refresh = HashSet::new();
+        for (_, id, name, _, _) in expired_translations {
+            if self
+                .permanent_translations
+                .values()
+                .filter_map(|(_, r_id, n, last_seen, slated_for_refresh)| {
+                    (r_id == id && *n == name).then_some((last_seen, slated_for_refresh))
+                })
+                .all(|(last_seen, slated_for_refresh)| {
+                    *slated_for_refresh || now.duration_since(*last_seen) > Duration::from_secs(30)
+                })
+            {
+                for_refresh.insert((name.clone(), self.id, *id));
+            }
+        }
+
+        for (name, conn_id, resource_id) in for_refresh {
+            self.buffered_events.push_back(GatewayEvent::RefreshDns {
```

Comment:
Would appreciate a log here with domain, IP and resource that we are refreshing.